lib/src/components/NxList/NxDescriptionListButtonItem.tsx

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { forwardRef } from 'react';
+import classnames from 'classnames';
+import { includesDisabledClass } from '../../util/classUtil';
+import { faAngleRight } from '@fortawesome/free-solid-svg-icons';
+import { NxFontAwesomeIcon } from '../..';
+import { NxDescriptionListButtonItemProps, nxDescriptionListButtonItemPropTypes} from './types';
+
+const NxDescriptionListButtonItem = forwardRef<HTMLDivElement, NxDescriptionListButtonItemProps>(
+    function NxDescriptionListButtonItem(props) {
+      const {
+            children,
+            className,
+            disabled,
+            selected: selectedProp,
+            buttonClassName,
+            buttonAttributes,
+            ...attrs
+          } = props,
+          selected = selectedProp ?? undefined,
+          divClassNames = classnames('nx-list__item', 'nx-list__item--clickable', className),
+          buttonClassNames = classnames('nx-list__btn', buttonClassName, { selected, disabled });
+
+      /* Using children[0].props.children allows for a more natural API for the consuming developer. */
```

Comment:
Unfortunately it makes more assumptions about the DOM that the caller is passing in.  For instance it assumes that they're passing in `<dt>My Content</dt>` and not `<MyCustomDt />`.  We try very hard not to make assumptions about JSX children wherever we can avoid it.  In this case I think a better API would be to create `NxDescriptionList.ButtonTerm` and `NxDescriptionList.ButtonDescription` components.  Or an an alternative, have `NxDescriptionList.ButtonItem` take the term and description JSX children as separate props instead of trying to dig them out of its own `children`.